src/Orchard.Web/Modules/Orchard.DynamicForms/ValidationRules/UrlAddress.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Text.RegularExpressions;
+using Orchard.DynamicForms.Helpers;
+using Orchard.DynamicForms.Services;
+using Orchard.DynamicForms.Services.Models;
+using Orchard.Localization;
+
+namespace Orchard.DynamicForms.ValidationRules
+{
+    public class UrlAddress : ValidationRule
+    {
+        public UrlAddress() {
+            RegexOptions = RegexOptions.Singleline | RegexOptions.IgnoreCase;
+            Pattern = @"^http(s)?://([\w-]+.)+[\w-]+(/[\w- ./?%&=])?$";
+        }
+
+        public string Pattern { get; set; }
+        public RegexOptions RegexOptions { get; set; }
+        
+        public override void Validate(ValidateInputContext context)
+        {
+            if (!Regex.IsMatch(context.AttemptedValue, Pattern, RegexOptions))
+            {
+                var message = GetValidationMessage(context);
+                context.ModelState.AddModelError(context.FieldName, message.Text);
+            }
+        }
+
+        public override void RegisterClientAttributes(RegisterClientValidationAttributesContext context)
+        {
+            context.ClientAttributes["data-val-regex"] = GetValidationMessage(context).Text;
+            context.ClientAttributes["data-val-regex-pattern"] = Pattern;
+        }
+
+        private LocalizedString GetValidationMessage(ValidationContext context)
+        {
+            return T(Tokenize(ErrorMessage.WithDefault(String.Format("{0} is not a valid url.", context.FieldName)), context));
```

Comment:
I still don't agree. If you call `T()` on something that is variable it will never work. Here `T` is called on `ErrorMessage` which comes from user input, or from `String.Format("{0} ...", fieldName)` with fieldName also coming from user input. So T can only be used on `"{0} ...", fieldName`.

```
    return String.IsNullOrEmpty(ErrorMessage) 
        ? T("{0} must match the value of {1}.", context.FieldName, TargetName)
        : new LocalizedString(Tokenize(ErrorMessage))
        ;
```